docs/data/pages.ts

```diff
@@ -277,41 +275,29 @@ const pages: MuiPage[] = [
             title: 'DateTimePickerToolbar',
           },
           { pathname: '/x/api/date-pickers/day-calendar-skeleton', title: 'DayCalendarSkeleton' },
+          { pathname: '/x/api/date-pickers/desktop-date-picker', title: 'DesktopDatePicker' },
           {
-            pathname: '/x/api/date-pickers/desktop-next-date-picker',
-            title: 'DesktopNextDatePicker',
-          },
-          {
-            pathname: '/x/api/date-pickers/desktop-next-date-range-picker',
-            title: 'DesktopNextDateRangePicker',
+            pathname: '/x/api/date-pickers/desktop-date-range-picker',
+            title: 'DesktopDateRangePicker',
             plan: 'pro',
           },
           {
-            pathname: '/x/api/date-pickers/desktop-next-date-time-picker',
-            title: 'DesktopNextDateTimePicker',
-          },
-          {
-            pathname: '/x/api/date-pickers/desktop-next-time-picker',
-            title: 'DesktopNextTimePicker',
+            pathname: '/x/api/date-pickers/desktop-date-time-picker',
+            title: 'DesktopDateTimePicker',
           },
+          { pathname: '/x/api/date-pickers/desktop-time-picker', title: 'DesktopTimePicker' },
           { pathname: '/x/api/date-pickers/localization-provider', title: 'LocalizationProvider' },
+          { pathname: '/x/api/date-pickers/mobile-date-picker', title: 'MobileDatePicker' },
           {
-            pathname: '/x/api/date-pickers/mobile-next-date-picker',
-            title: 'MobileNextDatePicker',
-          },
-          {
-            pathname: '/x/api/date-pickers/mobile-next-date-range-picker',
-            title: 'MobileNextDateRangePicker',
+            pathname: '/x/api/date-pickers/mobile-date-range-picker',
+            title: 'MobileDateRangePicker',
             plan: 'pro',
           },
           {
-            pathname: '/x/api/date-pickers/mobile-next-date-time-picker',
-            title: 'MobileNextDateTimePicker',
-          },
-          {
-            pathname: '/x/api/date-pickers/mobile-next-time-picker',
-            title: 'MobileNextTimePicker',
+            pathname: '/x/api/date-pickers/mobile-date-time-picker',
+            title: 'MobileDateTimePicker',
           },
+          { pathname: '/x/api/date-pickers/mobile-picker', title: 'MobileTimePicker' },
```

Comment:
404 fixed in https://github.com/mui/mui-x/commit/d2bd7eef3948b1b29bf2d4c5f16e21ede4268f1f. 

In the future, we could automate the generation of these pages. The Next.js page is generated in `buildComponentsDocumentation` so I think that we could also generate these API routes, like it's done in the main repo: https://github.com/mui/material-ui/blob/8ec664cae22b027d128363b15c9ed2c9dc4402b5/docs/data/base/pages.ts#L1